examples/bind/ecommerce.amp.html

```diff
@@ -48,53 +54,73 @@
     <amp-img width=50 height=50 src="./shirts/wine.jpg" option=8></amp-img>
   </amp-selector>
 
-  <amp-state id="shirts">
+  <!--
+    Available shirts, maps unique identifier to {color, image}. Sizes are fetched remotely.
+
+    When a shirt color is selected, `selectedIndex` changes which causes this
+    <amp-state> to perform an XHR to the new [src] value. Upon completion,
+    the XHR response is merged into the state under its "id" attribute ("shirts").
+
+    This has the limitation that data returned by the XHR must have the same
+    hierarchical structure as the existing data.
+
+    For example, the XHR must return something like:
+      {
+        "3": {
+          "sizes": ["S", "L"]
+        }
+      }
+
+    By contrast, it must NOT return something like:
+      {
+        "sizes": ["S", "L"]
+      }
+
+    In order to support the latter, we'd need to:
+      - Add support for a new action event, AND
+      - Force quotation marks for literal keys, AND
+      - Reintroduce variables in object keys (removed in #7582)
```

Comment:
I don't think this works for appending additional data to existing objects. We currently can add/remove elements from arrays but lack the ability to mutate array elements themselves. Perhaps we should figure out a way to support this though.